### DISCUSSION TITLE
Only file and data URLs are supported by the default ESM loader

### DISCUSSION THREAD

**Discussion by mattpodolak** [vercel/next.js#29080] [Help]
### What version of Next.js are you using?

11.1.2

### What version of Node.js are you using?

14.5.0

### What browser are you using?

Chrome

### What operating system are you using?

Windows

### How are you deploying your application?

local

### Describe the Bug

Performing a dynamic import using a URL triggers this error:

```
An error occured while importing fig module Error [ERR_UNSUPPORTED_ESM_URL_SCHEME]: Only file and data URLs are supported by the default ESM loader
```

There should at least be a feature to specify a custom loader: https://nodejs.org/api/esm.html#esm_https_loader

### Expected Behavior

To import module with no problem/error

### To Reproduce
```javascript
async () => {
const module = await import(
    /* webpackIgnore: true */
    `https://cdn.skypack.dev/@withfig/autocomplete/specs/git.js`
  );
}
```

**Comment by zh30** [comment]
I also had this problem, and finally I tried to upgrade the node version to 16.13.1, no such error occurred

**Comment by user72356** [comment]
I have the same problem with the latest version of node 18

**Reply by Ebrahim-Ramadan** [reply]
any approaches you took to overcome?
